zookeeper-server/src/main/java/org/apache/zookeeper/server/ContainerManager.java

```diff
@@ -129,8 +128,8 @@ public void checkContainers() throws InterruptedException {
         for (String containerPath : getCandidates()) {
             long startMs = Time.currentElapsedTime();
 
-            ByteBuffer path = ByteBuffer.wrap(containerPath.getBytes(UTF_8));
-            Request request = new Request(null, 0, 0, ZooDefs.OpCode.deleteContainer, path, null);
+            DeleteTxn txn = new DeleteTxn(containerPath);
+            Request request = new Request(null, 0, 0, ZooDefs.OpCode.deleteContainer, RequestRecord.fromRecord(txn), null);
```

Comment:
Well. The serialization logics are different from Jute's string and UTF-8 string. This can make it impossible rolling update as new server send new DeleteTxn but old server cannot recognize it.

Pending to revert.

Reply:
Reverted at a3ca9c41cbfc50ad08473c62ca3bfb2070de9269